hadoop-hdds/common/src/main/java/org/apache/hadoop/utils/db/RDBStore.java

```diff
@@ -318,6 +320,44 @@ public CodecRegistry getCodecRegistry() {
     return codecRegistry;
   }
 
+  @Override
+  public DBUpdatesWrapper getUpdatesSince(long sequenceNumber)
+      throws DataNotFoundException {
+
+    DBUpdatesWrapper dbUpdatesWrapper = new DBUpdatesWrapper();
+    try {
+      TransactionLogIterator transactionLogIterator =
+          db.getUpdatesSince(sequenceNumber);
+
+      boolean flag = true;
+
+      while (transactionLogIterator.isValid()) {
```

Comment:
Does this imply flush to sst can happen while iterating the log?

Reply:
yes, flushes can happen from memtables to SST anytime. If the WAL is deleted while it is being read, we will still be handle it through a retry mechanism from Recon side.

Reply:
So this means we need to at least configure the flushes defensively to maybe 3 * recon sync internal. Will that be a separate Jira or is already correctly configured?